Eufy widget crash

```
type: crash

SystemUptimeMs: 29724281
Process: system_server
Timestamp: 2026-02-26 07:05:45.156+0100
Build: google/caiman/caiman:16/BP4A.260205.002/2026021200:user/release-keys
Crash-Handler: com.android.internal.os.RuntimeInit$KillApplicationHandler
Loading-Progress: 1.0
Dropped-Count: 0

java.lang.RuntimeException: Unable to start activity ComponentInfo{android/android.appwidget.AppWidgetConfigActivityProxy}
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4514)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4717)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:224)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:133)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:103)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:80)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2970)
	at android.os.Handler.dispatchMessage(Handler.java:132)
	at android.os.Looper.dispatchMessage(Looper.java:333)
	at android.os.Looper.loopOnce(Looper.java:263)
	at android.os.Looper.loop(Looper.java:367)
	at com.android.server.SystemServer.run(SystemServer.java:1093)
	at com.android.server.SystemServer.main(SystemServer.java:716)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:933)
Caused by: android.content.ActivityNotFoundException: Unable to find explicit activity class {com.oceanwing.battery.cam/com.anker.clean_rn_bridge.app_widget.WidgetConfigActivity}; have you declared this activity in your AndroidManifest.xml, or does your intent not match its declared <intent-filter>?
	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2466)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:2031)
	at android.app.Activity.startActivityForResult(Activity.java:6104)
	at android.app.Activity.startActivityForResult(Activity.java:6023)
	at android.appwidget.AppWidgetConfigActivityProxy.onCreate(AppWidgetConfigActivityProxy.java:50)
	at android.app.Activity.performCreate(Activity.java:9359)
	at android.app.Activity.performCreate(Activity.java:9337)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1551)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4498)
	... 15 more
```

--- a/services/appwidget/java/com/android/server/appwidget/AppWidgetServiceImpl.java
+++ b/services/appwidget/java/com/android/server/appwidget/AppWidgetServiceImpl.java
@@ -3949,8 +3949,13 @@ class AppWidgetServiceImpl extends IAppWidgetService.Stub implements WidgetBacku
             String className = sa
                     .getString(com.android.internal.R.styleable.AppWidgetProviderInfo_configure);
             if (className != null) {
-                info.configure = new ComponentName(providerId.componentName.getPackageName(),
+                var cn = new ComponentName(providerId.componentName.getPackageName(),
                         className);
+                if (pm.resolveActivity(new Intent(AppWidgetManager.ACTION_APPWIDGET_CONFIGURE).setComponent(cn), 0) != null) {
+                    info.configure = cn;
+                } else {
+                    Slog.d(TAG, "ignoring invalid configuration activity: " + cn.toShortString());
+                }
             }
             info.label = activityInfo.loadLabel(pm).toString();
             info.icon = activityInfo.getIconResource();

